ansible/roles/postldap/defaults/main.yml

```diff
@@ -291,45 +292,63 @@ postldap__postfix__dependent_lookup_tables:
-      (configured in the attribute mail, aka the mailbox or maildrop).
+      The virtual_alias_maps setting is used to find the final delivery address
+      (configured in the attribute mailAddress).
     no_log: '{{ postldap__no_log|d(True) }}'
     connection: '{{ postldap__postfix_ldap_connection }}'
     search_base: '{{ postldap__ldap_people_dn|join(",") }}'
     query_filter: '(&
-                    (objectClass=inetOrgPerson)
-                    (|
-                      (mail=%s)
-                      (mailAlias=%s)
-                    )
-                    (|
-                      (authorizedService=all)
-                      (authorizedService=mail:receive)
-                    )
-                  )'
-    result_attribute: 'mail'
+                     (objectClass=inetOrgPerson)
+                     (|
+                       (mailAddress=%s)
+                       (mailAlternateAddress=%s)
+                     )
+                     (|
+                       (authorizedService=all)
+                       (authorizedService=mail:receive)
+                     )
+                   )'
+    result_attribute: 'mailAddress'
+
+  - name: 'ldap_virtual_groups_maps.cf'
+    state: 'present'
+    comment: |
+      The virtual_groups_maps setting is used to find the delivery addresses of
+      the members of a group (mailAlias or mailDistributionList) (configured in
+      the attribute mailForwardTo).
+    no_log: '{{ postldap__no_log|d(True) }}'
+    connection: '{{ postldap__postfix_ldap_connection }}'
+    search_base: '{{ postldap__ldap_groups_dn|join(",") }}'
+    query_filter: '(&
+                     (|
+                       (objectClass=mailAlias)
+                       (objectClass=mailDistributionList)
+                     )
+                     (mailAddress=%s)
+                   )'
+    result_attribute: 'mailForwardTo'
 
   - name: 'ldap_virtual_mailbox_maps.cf'
     state: 'present'
     comment: |
       As we only want to accept mail, where we know the recipients
       (and are responsible for them), the virtual_mailbox_maps configuration is used.
+      The path for the mailbox is retrieved from the mailHomeDirectory attribute
     no_log: '{{ postldap__no_log|d(True) }}'
     connection: '{{ postldap__postfix_ldap_connection }}'
     search_base: '{{ postldap__ldap_people_dn|join(",") }}'
     query_filter: '(&
                     (objectClass=inetOrgPerson)
                     (|
-                      (mail=%s)
-                      (mailAlias=%s)
+                      (mailAddress=%s)
+                      (mailAlternateAddress=%s)
                     )
                     (|
                       (authorizedService=all)
                       (authorizedService=mail:receive)
                     )
                   )'
-    result_attribute: 'mail'
-    result_format: '/%d/%u/Maildir/'
+    result_attribute: 'mailHomeDirectory'
```

Comment:
To have it we need to set two variables one for `result_attribute` and one for `result_format`. Am I right ?
In the documentation we could give examples to cover the three discussed possibilities:

- `/var/vmail/<uid>/Maildir `
- `/var/vmail/<domain>/<localpart>/Maildir `
- `<mailHomeDirectory>`

If you have any suggestion of how to do it properly I am open to suggestions.

Is it possible to have a mix ? If mailHomeDirectory is defined for the user, use it, otherwise use a scheme. I do not know enough of LDAP to know whether it is easy to implement.

Reply:
In theory you could have two separate Postfix lookup tables, one with `(mailHomeDirectory=*)`, another with a different filter. But that's probably too complex for our purposes, I would keep just two separate variables to define the attribute and format. As for the variable names, how about `postldap__virtual_mailbox_maps_attribute`, `postldap__virtual_mailbox_maps_format` with mention that these are specifically for the `ldap_virtual_mailbox_maps.cf` lookup table.

Reply:
Here it is, both as variables. As default, I kept imrejonk settings that might be more common.

Last question for this point : where do we set the corresponding documentation for the different use case ? I see at least three possibilities:
- in the `getting-started` page in a paragraph under "Overriding lookup table configuration";
- in the `guides` page
- close to the variable definition in a paragraph before/after.

Reply:
Thanks for updating the role, @tblein. Are the changes I'm proposing still compatible with your setup?

About documenting the mailbox location possibilities: I think these should go in the 'default variables' section. The getting started guide should be updated to reflect the new role defaults.

Reply:
@imrejonk  your changes are perfectly compatible with my setup. Thanks for your concern.

In the same time the final delivery of the emails on my system are not handled by postfix but by dovecot (through lmtp) to have the sieve filters. Therefore, the mail home is set in dovecot settings.

Reply:
That's good to hear!

Regarding LMTP delivery to Dovecot: we do that as well (for the same reason as you, sieve filtering), so I guess that the virtual_mailbox_maps setting in my main.cf doesn't actually do anything. See [this blog post](https://www.ciphermail.com/blog/eating-our-own-dog-food.html) if you're interested in the mail setup at my place of work.

We set Dovecot's mail_home like this: `dovecot_vmail_home: '/var/vmail/%u'`